ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * @file
+ * Contains helper functions for the support form.
+ */
+
+/**
+ * Search for a drupal user email in Jira.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ *
+ * @return bool|mixed
+ *   A jira user or FALSE if no user was found.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_search_jira_user($user, $form_state) {
+  $form_email = $form_state['values']['ereolen_support_email'];
+  $requestParts = _ereolen_support_get_request_parts('GET');
+  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $form_email, $requestParts['options']);
+  $resultArray = json_decode($result->data);
+  _ereolen_support_log_error($result);
+
+  // Searches for matches in full name, username and email, so we may have
+  // multiple results.
+  if(!empty($resultArray)) {
+    foreach ($resultArray as $jiraUser) {
+      if($jiraUser->name == $form_email) {
+        // If user is deactivated in Jira we need to activate it.
+        if ($jiraUser->active == false) {
+          $jiraUser = _ereolen_support_change_jira_user_status($jiraUser, $form_state);
+        }
+        return $jiraUser;
+      }
+    }
+  }
+
+  return FALSE;
+}
+
+/**
+ * Create a new Jira user.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ * @param array $form_state
+ *   The state of the form.
+ *
+ * @return mixed|null
+ *   A newly created Jira user or Null if no user was created.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_create_jira_user($user, $form_state) {
+  $form_email = $form_state['values']['ereolen_support_email'];
+  $data = [
+    'name' => $form_email,
+    'emailAddress' => $form_email,
+    'displayName' => $form_email,
+  ];
+
+  $requestParts = _ereolen_support_get_request_parts('POST', $data);
+
+  $result = drupal_http_request($requestParts['jira_path'] . '/user', $requestParts['options']);
+  _ereolen_support_log_error($result);
+  $userCreated = json_decode($result->data);
+  if (!empty($userCreated)) {
+    return $userCreated;
+  }
+
+  return NULL;
+}
+
+/**
+ * Set a Jira user to be active.
+ *
+ * @param \stdClass $jiraUser
+ *   A Jira user.
+ * @param array $form_state
+ *   The state of the form.
+ *
+ * @return mixed
+ *   A Jira user with status active.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_change_jira_user_status($jiraUser, $form_state) {
+  $data = [
```

Comment:
Would it be possible to check the users permissions in jira before activating it... the mail address is user inputted and can not be trusted.